### PR TITLE
Fix passing unclamped values to Color

### PR DESCRIPTION
**Pull request by Zyie**
Using a value for `this.alpha` on a Graphics object over 1 would crash the app

A common use case is using a tween to fade the alpha and depending on the ease the number can go above/below 0-1
```
await gsap.to(this, {
    alpha: 1, // alpha to 1
    ease: Back.easeOut.config(1.7),
});
```

### CODE REVIEW

**Review by GoodBoyDigital — APPROVED** [review]

File: packages/color/src/Color.ts [hunk 1]
```diff
@@ -471,7 +471,7 @@ export class Color
      */
     setAlpha(alpha: number): this
     {
-        this._components[3] = alpha;
+        this._components[3] = Math.min(1, Math.max(0, alpha));
```
Comment (bigtimebuddy) [inline packages/color/src/Color.ts:474]: Could you make a clamp function and reuse it here?

**Comment by Zyie** [comment]
@GoodBoyDigital @bigtimebuddy updated to override arrays instead of creating a new one

File: packages/color/src/Color.ts [hunk 2]
```diff
@@ -597,10 +602,31 @@ export class Color
...
         this._int = (((r * 255) << 16) + ((g * 255) << 8) + (b * 255 | 0));
     }
+
+    /**
+     * Clamps values to a range. Will override original values
+     * @param value - Value(s) to clamp
+     * @param min - Minimum value
+     * @param max - Maximum value
+     */
+    private _clamp<T extends number | number[] | ColorSourceTypedArray>(value: T, min: number, max: number): T
```
Comment (bigtimebuddy) [inline packages/color/src/Color.ts:618]:
```suggestion
    private _clamp<T extends number | number[] | ColorSourceTypedArray>(value: T, min: number = 0, max: number = 1): T
```

File: packages/color/src/Color.ts [hunk 3]
```diff
@@ -597,10 +602,31 @@ export class Color
     private refreshInt(): void
     {
         // Clamp values to 0 - 1
-        this._components.forEach((value, i) => (this._components[i] = Math.min(Math.max(value, 0), 1)));
+        this._clamp(this._components, 0, 1);
```
Comment (bigtimebuddy) [inline packages/color/src/Color.ts:605]:
```suggestion
        this._clamp(this._components);
```

File: packages/color/src/Color.ts [hunk 4]
```diff
@@ -542,10 +544,11 @@ export class Color
         }
         else if ((Array.isArray(value) || value instanceof Float32Array)
             // Can be rgb or rgba
-            && value.length >= 3 && value.length <= 4
-            // make sure all values are 0 - 1
-            && value.every((v) => v <= 1 && v >= 0))
+            && value.length >= 3 && value.length <= 4)
         {
+            // make sure all values are 0 - 1
+            value = this._clamp(value, 0, 1);
```
Comment (bigtimebuddy) [inline packages/color/src/Color.ts:550]:
```suggestion
            value = this._clamp(value);
```

File: packages/color/src/Color.ts [hunk 5]
```diff
@@ -471,7 +473,7 @@ export class Color
      */
     setAlpha(alpha: number): this
     {
-        this._components[3] = alpha;
+        this._components[3] = this._clamp(alpha, 0, 1);
```
Comment (bigtimebuddy) [inline packages/color/src/Color.ts:476]:
```suggestion
        this._components[3] = this._clamp(alpha);
```

**Review by bigtimebuddy — APPROVED** [review]
Minor suggestions to set min/max defaults. Otherwise looks good

**Pull request merged**